### PR TITLE
Converted MerkleAuditProof to a POJO.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.certificatetransparency.ctlog;
+
+import org.certificatetransparency.ctlog.proto.Ct;
+
+
+public class MerkleAuditProof {
+  public Ct.Version version;
+  public long treeSize;
+  public long leafIndex;
+  public java.util.List<byte[]> pathNode;
```
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java]: erm, wouldn't it be more readable to import java.util.List at the top?

File: java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  public MerkleAuditProof(Ct.Version version_, long treeSize_, long leafIndex_) {
```
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java]: this is odd style.  trailing underscores I mean.  you might remove them, and have the lines below read 'this.version = version' which is pretty much how I've seen Java done.

File: java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java (hunk 1 above)
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java]: I think I stole this from somewhere which had somehow decided that this was okay, but sure.

File: java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    version = version_;
+    treeSize = treeSize_;
+    leafIndex = leafIndex_;
+    pathNode = new java.util.ArrayList<byte[]>();
```
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java]: Is that a reasonable implementation of `List<>` to use? I just looked around and poked in the dark until it compiled and the tests passed, really... :wink:

File: java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java (hunk 2 above)
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java]: Done.

File: java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java (hunk 3 above)
Comment (phad) [inline java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java]: Yes, I think it's ok.

File: java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  public Ct.Version version;
+  public long treeSize;
+  public long leafIndex;
+  public List<byte[]> pathNode;
+
+  public MerkleAuditProof(Ct.Version version, long treeSize, long leafIndex) {
+    this.version = version;
+    this.treeSize = treeSize;
+    this.leafIndex = leafIndex;
+    pathNode = new ArrayList<byte[]>();
```
Comment (pphaneuf) [inline java/src/org/certificatetransparency/ctlog/MerkleAuditProof.java]: Is that a reasonable implementation of `List<>` to use? I just looked around and poked in the dark until it compiled and the tests passed, really... :wink:

**Pull request merged**